sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -881,7 +881,16 @@ class Analyzer(
             }
           case u @ UnresolvedFunction(funcId, children, isDistinct) =>
             withPosition(u) {
-              catalog.lookupFunction(funcId, children) match {
+              val func = try {
+                catalog.lookupFunction(funcId, children)
+              } catch {
+                // SPARK-16228 ExternalCatalog may recognize `double`-type only.
+                case _: Exception =>
+                  val newChildren = children.map(x =>
```

Comment:
I am not sure. That seems like a good place though.

I am also curious why we use the function arguments (and not just the names) to resolve the function; this kinda robs us from the opportunity to cast the arguments.